### DISCUSSION TITLE
Is there a way to comment out some code in XAML that only executes in debug mode

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15692] [Q&A]
Similar to

```
#if DEBUG
    DoSomething();
#endif
```

in C#?

**Comment by thevortexcloud** [comment]
Sort of? You can use the various design time stuff. You can also make a custom mark up extension that just does nothing if in debug mode/returns a true/false value which can be used to set the visibility on things. You could also probably make a custom content control that renders nothing if it's in debug. But I have not tried that. 

XAML is as the name implies, a mark up language. This sort of thing is generally done in the code behind.

**Reply by workgroupengineering** [reply]
I think he meant something like [this](https://learn.microsoft.com/en-us/windows/uwp/debug-test-perf/conditional-xaml) or [this](https://github.com/firstfloorsoftware/xcc/wiki).
